src/xbot/common/injection/wpi_factories/RealWPIFactory.java

```diff
@@ -98,14 +100,24 @@ public AdvancedJoystickButton getJoystickButton(XJoystick joystick, int button_n
     }
 
     @Override
-    public XGyro getGyro() {
+    public XGyro getGyro(ImuType imuType) {
         // It's possible that the nav6 might get disconnected, and throw some
         // exceptions
         // at runtime when it can't communicate over the serial port.
         // The robot needs to protect itself from this behavior.
         try {
-            return new Nav6Gyro();
+            if (imuType == ImuType.nav6) {
```

Comment:
A little cleaner as a case statement maybe? Or if you want if's, make the subsequent ones else ifs just to be explicit?